src/inmanta/execute/dataflow/datatrace.py

```diff
@@ -0,0 +1,241 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from itertools import chain
+from typing import TYPE_CHECKING, Iterable, List, Optional
+
+from inmanta.ast import Locatable, NotFoundException
+from inmanta.ast.statements import Statement
+from inmanta.execute.dataflow import (
+    AssignableNode,
+    AssignableNodeReference,
+    Assignment,
+    AttributeNode,
+    AttributeNodeReference,
+    DataflowGraph,
+    Equivalence,
+    InstanceNode,
+    InstanceNodeReference,
+    VariableNodeReference,
+)
+from inmanta.execute.runtime import Instance
+
+if TYPE_CHECKING:
+    pass
+
+
+class DataTraceRenderer:
```

Comment:
Please document the main entry point, what this class does (general overview of the function)